MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpress.h

```diff
@@ -161,6 +177,18 @@ struct _NVME_CONTROLLER_PRIVATE_DATA {
   //
   NVME_ADMIN_CONTROLLER_DATA            *ControllerData;
 
+  //
+  // Number of Queues Allocated by the controller
+  //
+  UINT32                                Nsqa; // Number of Submission Queues Allocated

```

Comment:
The above checks are only checking Nsqa.

Uefi always uses a 1:1 submission/completion queue allocation. Is there any benefit to having both Ncqa and Nsqa? Would it be better to just track a single SubmissionCompletionPairCount?

Reply:
Because there can be multiple submission queues for 1 completion queue, and the possibility of that scenario doesn't seem far off, it made sense to me to check only Nsqa. Since in that case we would have 1 completion queue, 1 synchronous submission queue, and 1 asynchronous submission queue. 
The goal here was to be more flexible for future scenarios. It may be unnecessary for this PR, though. Will start a private thread for discussion.

Reply:
@apop5  Changed to `NumberOfDataQueuePairs` since the consensus was, we only support 1:1 submission: completion queues.